### ISSUE TITLE
MSVC/Visual Studio 2019: Linking errors

### ISSUE THREAD

**Issue by Vironacorus** [KhronosGroup/glslang#2506]
After compiling glslang using build instructions provided (CMake), I tried linking glslang to my project. I'm getting lots of unresolved externals, though. Mostly inside of SPIRVd.lib.

These are libraries I link to, in that order:
glslangd.lib
HLSLd.lib
OSDependentd.lib
OGLCompilerd.lib
SPIRVd.lib

It's important to note that I use C and not C++.
Does someone know how to fix this problem?

**Comment by greg-lunarg** [comment]
Can you give the unresolved externals?

What commit are you trying to build with? What was the last commit you were able to build with?

**Comment by greg-lunarg** [comment]
What platform are you building on? Order is important on Linux.

**Comment by Vironacorus** [comment]
@greg-lunarg
1. Wait a second 
2. I'm not sure what commit, I just git cloned.
3. Windows.

**Comment by greg-lunarg** [comment]
Here are the libs and order that are currently used by the SDK to test the libs it delivers for Linux and other:

if (CMAKE_SYSTEM_NAME MATCHES "Linux")
    set (PROJECT_LINK_LIBS  glslang SPIRV HLSL MachineIndependent OGLCompiler OSDependent GenericCodeGen SPIRV-Tools-opt SPIRV-Tools SPVRemapper pthread)
else ()
    set (PROJECT_LINK_LIBS glslang OGLCompiler SPIRV HLSL OSDependent MachineIndependent GenericCodeGen SPIRV-Tools-opt SPIRV-Tools SPVRemapper)
endif()

I am really not sure why the Windows order is different from Linux. Last I saw order doesn't matter on Windows; I believe cmake figures out the right order on Windows. If you think the order is the issue, I would try the Linux order.

I suspect that your problem is not order, but that you are missing MachineIndependent and GenericCodeGen which were added recently. I will check that the build instructions in the README are up to date.

You might not need SPIRV-Tools-opt, SPIRV-Tools, and SPVRemapper, so don't worry about those right away.
